ignite/engine/__init__.py

```diff
@@ -95,6 +95,8 @@ def supervised_training_step(
         )
 
     def update(engine: Engine, batch: Sequence[torch.Tensor]) -> Union[Any, Tuple[torch.Tensor]]:
+        if (engine.state.iteration - 1) % gradient_accumulation_steps == 0:
```

Comment:
@sadra-barikbin I prefer to finalize already open PR. Can you make another branch and use commits from https://github.com/pytorch/ignite/pull/2470 and add tests I asked in the PR. Such that we merge your PR first and update this one right after that.